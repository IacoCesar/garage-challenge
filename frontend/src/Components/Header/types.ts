export interface IHeader {
  testID?: string
  isHomeScreen?: boolean
  title?: string
  callHandleBack?: (navigation: any) => void
  callHandleChangeVisualMode?: (themeSelector: string) => void
}

export interface IProps extends IHeader {
  themeSelector: string
}
