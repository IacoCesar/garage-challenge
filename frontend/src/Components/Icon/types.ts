export interface IProps {
  testID?: string
  size: number
  name: string
  color: string
  type?: string
}
