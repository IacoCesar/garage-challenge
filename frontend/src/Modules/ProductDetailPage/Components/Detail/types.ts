export interface IProps {
  testID?: string
}

export interface IThemeProps {
  themeSelector: string
}
