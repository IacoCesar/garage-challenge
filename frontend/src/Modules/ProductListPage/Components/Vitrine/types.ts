export interface IFavoriteCarProps {
  carId: string
}

export interface IRenderItemProps {
  item: {
    id: string
    model: string
    make: string
    year: number
    image: {
      url: string
    }
  }
  index: number
}

export interface IProps {
  testID?: string
  isLoading: boolean
  vitrineData: object[]
  favoriteCarsData: IFavoriteCarProps[]
  callHandleOpenPdp: (item: object) => void
  callHandleFavoriteCar: (id: string) => void
  callHandleUnFavoriteCar: (id: string) => void
  callLoadVitrineData: () => void
}

export interface IThemeProps {
  themeSelector: string
}
