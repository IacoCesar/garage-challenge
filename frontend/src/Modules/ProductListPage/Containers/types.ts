export type IRenderItemProps = {
  item: {
    id: string
    model: string
    make: string
    year: number
    image: {
      url: string
    }
  }
  index: number
}
