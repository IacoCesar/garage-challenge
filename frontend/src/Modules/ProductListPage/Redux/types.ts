export interface IPayloadProps {
  id: string
  model: string
  make: string
  year: number
  image: {
    url: string
  }
}

export interface IVitrineState {
  vitrineItemSelector: IPayloadProps
}
