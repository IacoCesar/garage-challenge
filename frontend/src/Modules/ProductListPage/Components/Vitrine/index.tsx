import React, { useCallback } from 'react'
import S from './styles'
import { IProps, IRenderItemProps, IFavoriteCarProps } from './types'
import Text from '@/Components/Text'
import Spacer from '@/Components/Spacer'
import LoaderContent from '@/Components/LoaderContent'
import Icon from '@/Components/Icon'
import { useSelector } from 'react-redux'
import { RootState } from '@/Store'
import { useTheme } from 'styled-components/native'
import { assignTestId } from '@/Helpers/TestId'
import { API_BASE } from '@env'

const Vitrine: React.FC<IProps> = ({
  testID = 'Vitrine',
  isLoading,
  vitrineData = [],
  favoriteCarsData = [],
  callHandleOpenPdp,
  callHandleFavoriteCar,
  callHandleUnFavoriteCar,
  callLoadVitrineData
}) => {
  const themeSelector = useSelector((state: RootState) => state.themeData.theme)
  const theme = useTheme()

  const _callRenderItem: any = useCallback(
    ({ item, index }: IRenderItemProps) => {
      const filterFavoriteCar = favoriteCarsData.filter(
        (it: IFavoriteCarProps) => it.carId === item.id
      )
      const isFavoritedCar = filterFavoriteCar.length > 0

      return (
        <S.ContentListItem
          {...assignTestId(
            'TouchableOpacity',
            `${testID}_Content_List_ContentListItem_${index}`
          )}
          style={{
            shadowColor: theme.standard[themeSelector].shadow,
            shadowOffset: { width: -2, height: 4 },
            shadowOpacity: 0.2,
            shadowRadius: 3
          }}
          key={index}
          onPress={() => callHandleOpenPdp(item)}
          themeSelector={themeSelector}
        >
          <S.ContentListImage
            {...assignTestId(
              'Image',
              `${testID}_Content_List_ContentListItem_${index}_ContentListImage`
            )}
            themeSelector={themeSelector}
            source={{ uri: `${API_BASE}${item.image.url}` }}
            resizeMode="cover"
          />
          <S.ContentListInfo
            {...assignTestId(
              'View',
              `${testID}_Content_List_ContentListItem_${index}_ContentListInfo`
            )}
          >
            <Spacer
              testID={`${testID}_Content_List_ContentListItem_${index}_ContentListInfo_Spacer`}
              min={10}
              max={10}
            />
            <S.ContentListInfoTop
              {...assignTestId(
                'View',
                `${testID}_Content_List_ContentListItem_${index}_ContentListInfo_ContentListInfoTop`
              )}
            >
              <S.ContentListInfoTopDesc
                {...assignTestId(
                  'View',
                  `${testID}_Content_List_ContentListItem_${index}_ContentListInfo_ContentListInfoTop_ContentListInfoTopDesc`
                )}
              >
                <Text
                  testID={`${testID}_Content_List_ContentListItem_${index}_ContentListInfo_ContentListInfoTop_ContentListInfoTopDesc_Text`}
                  fontSize={23}
                  label={item.model}
                  color={theme.standard[themeSelector].cardTextColor}
                />
              </S.ContentListInfoTopDesc>
              <S.ContentListInfoTopStar
                {...assignTestId(
                  'TouchableOpacity',
                  `${testID}_Content_List_ContentListItem_${index}_ContentListInfo_ContentListInfoTop_ContentListInfoTopStar`
                )}
                onPress={() =>
                  isFavoritedCar
                    ? callHandleUnFavoriteCar(item.id)
                    : callHandleFavoriteCar(item.id)
                }
              >
                <Icon
                  testID={`${testID}_Content_List_ContentListItem_${index}_ContentListInfo_ContentListInfoTop_ContentListInfoTopStar_Icon`}
                  size={25}
                  name={isFavoritedCar ? 'star' : 'staro'}
                  color={
                    isFavoritedCar
                      ? theme.colors.starColor
                      : theme.standard.default.textColor
                  }
                />
              </S.ContentListInfoTopStar>
            </S.ContentListInfoTop>
            <Spacer
              testID={`${testID}_Content_List_ContentListItem_${index}_ContentListInfo_Spacer`}
              min={10}
              max={10}
            />
            <S.ContentListHr
              {...assignTestId(
                'View',
                `${testID}_Content_List_ContentListItem_${index}_ContentListInfo_ContentListHr`
              )}
              themeSelector={themeSelector}
            />
            <Spacer
              testID={`${testID}_Content_List_ContentListItem_${index}_ContentListInfo_Spacer2`}
              min={10}
              max={10}
            />
            <S.ContentListInfoBot
              {...assignTestId(
                'View',
                `${testID}_Content_List_ContentListItem_${index}_ContentListInfo_ContentListInfoBot`
              )}
            >
              <S.ContentListInfoBotDesc
                {...assignTestId(
                  'View',
                  `${testID}_Content_List_ContentListItem_${index}_ContentListInfo_ContentListInfoBot_ContentListInfoBotDesc`
                )}
              >
                <Text
                  testID={`${testID}_Content_List_ContentListItem_${index}_ContentListInfo_ContentListInfoBot_ContentListInfoBotDesc_Text`}
                  label={`${item.make} | ${item.year}`}
                  color={theme.standard[themeSelector].cardTextColor}
                />
              </S.ContentListInfoBotDesc>
            </S.ContentListInfoBot>
            <Spacer
              testID={`${testID}_Content_List_ContentListItem_${index}_ContentListInfo_Spacer`}
              min={10}
              max={10}
            />
          </S.ContentListInfo>
        </S.ContentListItem>
      )
    },
    [favoriteCarsData, vitrineData, themeSelector]
  )

  return (
    <>
      {isLoading && (
        <LoaderContent
          testID={`${testID}_LoaderContent`}
          size={25}
          color={theme.standard.default.background}
        />
      )}
      <S.Content testID={`${testID}_Content`}>
        <Spacer
          testID={`${testID}_Content_Spacer`}
          min={theme.spacing.global.horizontal}
          max={theme.spacing.global.horizontal}
        />
        <S.List
          testID={`${testID}_Content_List`}
          data={vitrineData}
          ItemSeparatorComponent={() => <Spacer min={20} max={20} />}
          keyExtractor={(_, index): string => index.toString()}
          renderItem={_callRenderItem}
          onEndReachedThreshold={0.5}
          onEndReached={callLoadVitrineData}
        />
        <Spacer
          testID={`${testID}_Content_Spacer_2`}
          min={theme.spacing.global.horizontal}
          max={theme.spacing.global.horizontal}
        />
      </S.Content>
    </>
  )
}

export default Vitrine
