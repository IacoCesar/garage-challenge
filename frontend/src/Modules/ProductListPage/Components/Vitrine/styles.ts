import styled from 'styled-components/native'
import { Dimensions } from 'react-native'
import VirtualizedScrollView from '@/Components/VirtualizedScrollView'
import { IThemeProps } from './types'

const screenW = Dimensions.get('screen').width

const Content = styled(VirtualizedScrollView)``

const List = styled.FlatList`
  padding-horizontal: ${({ theme }) => theme.spacing.global.horizontal};
`

const ContentListItem = styled.TouchableOpacity<IThemeProps>`
  display: flex;
  flex-direction: column;
  min-height: 300px;
  background-color: ${({ theme, themeSelector }) =>
    theme.standard[themeSelector].card};
`

const ContentListImage = styled.Image<IThemeProps>`
  display: flex;
  width: 100%;
  height: ${(screenW * 9) / 16};
  background-color: ${({ theme, themeSelector }) =>
    theme.standard[themeSelector].divisor};
`

const ContentListInfo = styled.View`
  display: flex;
  flex-direction: column;
`

const ContentListInfoTop = styled.View`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding-horizontal: ${({ theme }) => theme.spacing.global.horizontal};
`

const ContentListInfoTopDesc = styled.View`
  display: flex;
  flex-direction: row;
`

const ContentListInfoTopStar = styled.TouchableOpacity`
  display: flex;
  width: 25px;
  height: 25px;
  justify-content: center;
  align-items: center;
  background-color: transparent;
`

const ContentListHr = styled.View<IThemeProps>`
  display: flex;
  width: 100%;
  border-width: 2px;
  border-color: ${({ theme, themeSelector }) =>
    theme.standard[themeSelector].divisor};
`

const ContentListInfoBot = styled.View`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  padding-horizontal: ${({ theme }) => theme.spacing.global.horizontal};
`

const ContentListInfoBotDesc = styled.View`
  display: flex;
  align-items: center;
`

export default {
  Content,
  List,
  ContentListItem,
  ContentListHr,
  ContentListImage,
  ContentListInfo,
  ContentListInfoTop,
  ContentListInfoTopDesc,
  ContentListInfoTopStar,
  ContentListInfoBot,
  ContentListInfoBotDesc
}
