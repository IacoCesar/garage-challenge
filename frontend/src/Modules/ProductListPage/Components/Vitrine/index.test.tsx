import React from 'react'
import { IProps } from './types'
import { mockTestId } from '@/Helpers/TestId'
import Vitrine from '.'
import { render, fireEvent, act, waitFor } from '@/Helpers/TestHelpers'

describe('Vitrine', () => {
  let props: IProps = {
    testID: 'Vitrine',
    isLoading: false,
    favoriteCarsData: [
      {
        carId: '155fa5ae-00a0-11ea-84e3-4371baa68b28'
      }
    ],
    vitrineData: [
      {
        id: '206b3cd8-00a0-11ea-a1de-278776d54145',
        model: 'A6 50',
        make: 'Audi',
        year: 2013,
        image: {
          url: '/assets/vehicle_image-206b3cd8-00a0-11ea-a1de-278776d54145.jpg'
        }
      },
      {
        id: '155fa5ae-00a0-11ea-84e3-4371baa68b28',
        model: 'RS7 4.0',
        make: 'Audi',
        year: 2015,
        image: {
          url: '/assets/vehicle_image-155fa5ae-00a0-11ea-84e3-4371baa68b28.jpg'
        }
      }
    ],
    callHandleOpenPdp: jest.fn(),
    callHandleFavoriteCar: jest.fn(),
    callHandleUnFavoriteCar: jest.fn(),
    callLoadVitrineData: jest.fn()
  }

  const containerElement = mockTestId('List', `${props.testID!}_Content`).testID
  const cardElement = mockTestId(
    'TouchableOpacity',
    `${props.testID!}_Content_List_ContentListItem_0`
  ).testID

  const favoriteElement = mockTestId(
    'TouchableOpacity',
    `${props.testID!}_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoTop_ContentListInfoTopStar`
  ).testID

  const unFavoriteElement = mockTestId(
    'TouchableOpacity',
    `${props.testID!}_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoTop_ContentListInfoTopStar`
  ).testID

  it(`should render ${containerElement}`, async () => {
    const wrapper = render(<Vitrine {...props} />)

    const el_container = wrapper.getByTestId(containerElement)

    await waitFor(() => {
      expect(el_container).toBeTruthy()
      expect(el_container.children.length).toBe(1)
    })
  })

  it(`should onPress ${cardElement}`, async () => {
    const wrapper = render(<Vitrine {...props} />)

    const button = await wrapper.findByTestId(cardElement)

    await act(async () => {
      fireEvent.press(button)
      expect(props.callHandleOpenPdp).toHaveBeenCalled()
    })
  })

  it(`should onPress ${favoriteElement}`, async () => {
    const wrapper = render(<Vitrine {...props} />)

    const button = await wrapper.findByTestId(favoriteElement)

    await act(async () => {
      fireEvent.press(button)

      expect(props.callHandleFavoriteCar).toHaveBeenCalled()
    })
  })

  it(`should onPress ${favoriteElement}`, async () => {
    const wrapper = render(<Vitrine {...props} />)

    const button = await wrapper.findByTestId(favoriteElement)

    await act(async () => {
      fireEvent.press(button)
      expect(props.callHandleFavoriteCar).toHaveBeenCalled()
    })
  })

  it(`should onPress ${unFavoriteElement}`, async () => {
    const wrapper = render(<Vitrine {...props} />)

    const button = await wrapper.findByTestId(unFavoriteElement)

    await act(async () => {
      fireEvent.press(button)
      expect(props.callHandleUnFavoriteCar).toHaveBeenCalled()
    })
  })

  it('should render snapshot', async () => {
    const wrapper = render(<Vitrine {...props} />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
