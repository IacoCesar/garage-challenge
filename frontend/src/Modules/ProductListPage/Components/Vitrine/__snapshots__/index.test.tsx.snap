// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vitrine should render snapshot 1`] = `
<RCTScrollView
  ListHeaderComponent={
    <React.Fragment>
      <SpacerView
        max={16}
        min={16}
        testID="Vitrine_Content_Spacer"
      />
      <Styled(FlatList)
        ItemSeparatorComponent={[Function]}
        data={
          Array [
            Object {
              "id": "206b3cd8-00a0-11ea-a1de-278776d54145",
              "image": Object {
                "url": "/assets/vehicle_image-206b3cd8-00a0-11ea-a1de-278776d54145.jpg",
              },
              "make": "Audi",
              "model": "A6 50",
              "year": 2013,
            },
            Object {
              "id": "155fa5ae-00a0-11ea-84e3-4371baa68b28",
              "image": Object {
                "url": "/assets/vehicle_image-155fa5ae-00a0-11ea-84e3-4371baa68b28.jpg",
              },
              "make": "Audi",
              "model": "RS7 4.0",
              "year": 2015,
            },
          ]
        }
        keyExtractor={[Function]}
        onEndReached={[MockFunction]}
        onEndReachedThreshold={0.5}
        renderItem={[Function]}
        testID="Vitrine_Content_List"
      />
      <SpacerView
        max={16}
        min={16}
        testID="Vitrine_Content_Spacer_2"
      />
    </React.Fragment>
  }
  accessible={false}
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="List_Vitrine_Content"
  themeSelector="default"
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        accessible={false}
        max={16}
        min={16}
        style={
          Array [
            Object {
              "flexGrow": 1,
              "maxHeight": 16,
              "minHeight": 16,
            },
          ]
        }
        testID="View_Vitrine_Content_Spacer"
      />
      <View
        ItemSeparatorComponent={[Function]}
        data={
          Array [
            Object {
              "id": "206b3cd8-00a0-11ea-a1de-278776d54145",
              "image": Object {
                "url": "/assets/vehicle_image-206b3cd8-00a0-11ea-a1de-278776d54145.jpg",
              },
              "make": "Audi",
              "model": "A6 50",
              "year": 2013,
            },
            Object {
              "id": "155fa5ae-00a0-11ea-84e3-4371baa68b28",
              "image": Object {
                "url": "/assets/vehicle_image-155fa5ae-00a0-11ea-84e3-4371baa68b28.jpg",
              },
              "make": "Audi",
              "model": "RS7 4.0",
              "year": 2015,
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReached={[MockFunction]}
        onEndReachedThreshold={0.5}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Array [
            Object {
              "paddingHorizontal": 16,
            },
          ]
        }
        testID="Vitrine_Content_List"
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#ffffff",
                "display": "flex",
                "flexDirection": "column",
                "minHeight": 300,
                "opacity": 1,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": -2,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 3,
              }
            }
            testID="TouchableOpacity_Vitrine_Content_List_ContentListItem_0"
          >
            <Image
              accessible={false}
              resizeMode="cover"
              source={
                Object {
                  "uri": "http://localhost:3000/assets/vehicle_image-206b3cd8-00a0-11ea-a1de-278776d54145.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "backgroundColor": "#9F9F9F",
                    "display": "flex",
                    "height": 421.875,
                    "width": "100%",
                  },
                ]
              }
              testID="Image_Vitrine_Content_List_ContentListItem_0_ContentListImage"
              themeSelector="default"
            />
            <View
              accessible={false}
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                  },
                ]
              }
              testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo"
            >
              <View
                accessible={false}
                max={10}
                min={10}
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "maxHeight": 10,
                      "minHeight": 10,
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_Spacer"
              />
              <View
                accessible={false}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingHorizontal": 16,
                      "width": "100%",
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoTop"
              >
                <View
                  accessible={false}
                  style={
                    Array [
                      Object {
                        "display": "flex",
                        "flexDirection": "row",
                      },
                    ]
                  }
                  testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoTop_ContentListInfoTopDesc"
                >
                  <View
                    accessible={false}
                    style={
                      Array [
                        Object {
                          "display": "flex",
                        },
                      ]
                    }
                    testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoTop_ContentListInfoTopDesc_Text"
                  >
                    <Text
                      accessible={false}
                      color="#000000"
                      fontSize={23}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 23,
                          },
                        ]
                      }
                      testID="Text_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoTop_ContentListInfoTopDesc_Text_Content_Text"
                    >
                      A6 50
                    </Text>
                  </View>
                </View>
                <View
                  accessible={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "display": "flex",
                      "height": 25,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 25,
                    }
                  }
                  testID="TouchableOpacity_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoTop_ContentListInfoTopStar"
                >
                  <View
                    accessible={false}
                    color="#000000"
                    name="staro"
                    size={25}
                    testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoTop_ContentListInfoTopStar_Icon"
                  />
                </View>
              </View>
              <View
                accessible={false}
                max={10}
                min={10}
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "maxHeight": 10,
                      "minHeight": 10,
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_Spacer"
              />
              <View
                accessible={false}
                style={
                  Array [
                    Object {
                      "borderColor": "#9F9F9F",
                      "borderWidth": 2,
                      "display": "flex",
                      "width": "100%",
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListHr"
                themeSelector="default"
              />
              <View
                accessible={false}
                max={10}
                min={10}
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "maxHeight": 10,
                      "minHeight": 10,
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_Spacer2"
              />
              <View
                accessible={false}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "paddingHorizontal": 16,
                      "width": "100%",
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoBot"
              >
                <View
                  accessible={false}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                      },
                    ]
                  }
                  testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoBot_ContentListInfoBotDesc"
                >
                  <View
                    accessible={false}
                    style={
                      Array [
                        Object {
                          "display": "flex",
                        },
                      ]
                    }
                    testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoBot_ContentListInfoBotDesc_Text"
                  >
                    <Text
                      accessible={false}
                      color="#000000"
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 12,
                          },
                        ]
                      }
                      testID="Text_Vitrine_Content_List_ContentListItem_0_ContentListInfo_ContentListInfoBot_ContentListInfoBotDesc_Text_Content_Text"
                    >
                      Audi | 2013
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessible={false}
                max={10}
                min={10}
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "maxHeight": 10,
                      "minHeight": 10,
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_0_ContentListInfo_Spacer"
              />
            </View>
          </View>
          <View
            accessible={false}
            max={20}
            min={20}
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "maxHeight": 20,
                  "minHeight": 20,
                },
              ]
            }
            testID="View_Spacer"
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#ffffff",
                "display": "flex",
                "flexDirection": "column",
                "minHeight": 300,
                "opacity": 1,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": -2,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 3,
              }
            }
            testID="TouchableOpacity_Vitrine_Content_List_ContentListItem_1"
          >
            <Image
              accessible={false}
              resizeMode="cover"
              source={
                Object {
                  "uri": "http://localhost:3000/assets/vehicle_image-155fa5ae-00a0-11ea-84e3-4371baa68b28.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "backgroundColor": "#9F9F9F",
                    "display": "flex",
                    "height": 421.875,
                    "width": "100%",
                  },
                ]
              }
              testID="Image_Vitrine_Content_List_ContentListItem_1_ContentListImage"
              themeSelector="default"
            />
            <View
              accessible={false}
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                  },
                ]
              }
              testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo"
            >
              <View
                accessible={false}
                max={10}
                min={10}
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "maxHeight": 10,
                      "minHeight": 10,
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_Spacer"
              />
              <View
                accessible={false}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingHorizontal": 16,
                      "width": "100%",
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoTop"
              >
                <View
                  accessible={false}
                  style={
                    Array [
                      Object {
                        "display": "flex",
                        "flexDirection": "row",
                      },
                    ]
                  }
                  testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoTop_ContentListInfoTopDesc"
                >
                  <View
                    accessible={false}
                    style={
                      Array [
                        Object {
                          "display": "flex",
                        },
                      ]
                    }
                    testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoTop_ContentListInfoTopDesc_Text"
                  >
                    <Text
                      accessible={false}
                      color="#000000"
                      fontSize={23}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 23,
                          },
                        ]
                      }
                      testID="Text_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoTop_ContentListInfoTopDesc_Text_Content_Text"
                    >
                      RS7 4.0
                    </Text>
                  </View>
                </View>
                <View
                  accessible={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "display": "flex",
                      "height": 25,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 25,
                    }
                  }
                  testID="TouchableOpacity_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoTop_ContentListInfoTopStar"
                >
                  <View
                    accessible={false}
                    color="#F9E26B"
                    name="star"
                    size={25}
                    testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoTop_ContentListInfoTopStar_Icon"
                  />
                </View>
              </View>
              <View
                accessible={false}
                max={10}
                min={10}
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "maxHeight": 10,
                      "minHeight": 10,
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_Spacer"
              />
              <View
                accessible={false}
                style={
                  Array [
                    Object {
                      "borderColor": "#9F9F9F",
                      "borderWidth": 2,
                      "display": "flex",
                      "width": "100%",
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListHr"
                themeSelector="default"
              />
              <View
                accessible={false}
                max={10}
                min={10}
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "maxHeight": 10,
                      "minHeight": 10,
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_Spacer2"
              />
              <View
                accessible={false}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "paddingHorizontal": 16,
                      "width": "100%",
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoBot"
              >
                <View
                  accessible={false}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                      },
                    ]
                  }
                  testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoBot_ContentListInfoBotDesc"
                >
                  <View
                    accessible={false}
                    style={
                      Array [
                        Object {
                          "display": "flex",
                        },
                      ]
                    }
                    testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoBot_ContentListInfoBotDesc_Text"
                  >
                    <Text
                      accessible={false}
                      color="#000000"
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 12,
                          },
                        ]
                      }
                      testID="Text_Vitrine_Content_List_ContentListItem_1_ContentListInfo_ContentListInfoBot_ContentListInfoBotDesc_Text_Content_Text"
                    >
                      Audi | 2015
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessible={false}
                max={10}
                min={10}
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "maxHeight": 10,
                      "minHeight": 10,
                    },
                  ]
                }
                testID="View_Vitrine_Content_List_ContentListItem_1_ContentListInfo_Spacer"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={false}
        max={16}
        min={16}
        style={
          Array [
            Object {
              "flexGrow": 1,
              "maxHeight": 16,
              "minHeight": 16,
            },
          ]
        }
        testID="View_Vitrine_Content_Spacer_2"
      />
    </View>
  </View>
</RCTScrollView>
`;
